### ISSUE TITLE
Provide an example of output using Public Git Archive

### ISSUE THREAD

**Issue by campoy** [src-d/gemini#138]
I just downloaded all of the siva files in PGA corresponding to the Google, Docker, and *one more org* and ran Gemini using docker-compose.

The `hash` process fails at the end with this message:

```bash
ERROR 19:13:53 org.apache.spark.internal.Logging$class (Logging.scala:70) - SparkListenerBus
 has already stopped! Dropping event SparkListenerJobEnd(2,1529522033360,JobFailed(org.apach
e.spark.SparkException: Job 2 cancelled because SparkContext was shut down))
```

This is after successfully storing all of the information to DB, apparently.

```bash
 WARN 19:13:38 tech.sourced.gemini.Gemini (Gemini.scala:33) - Getting repositories at /repos
itories in siva format
 WARN 19:13:40 tech.sourced.gemini.Gemini (Gemini.scala:42) - Hashing
 WARN 19:13:40 tech.sourced.gemini.Hash (Hash.scala:56) - Listing files
 WARN 19:13:41 tech.sourced.gemini.Hash (Hash.scala:67) - Extracting UASTs
 WARN 19:13:42 tech.sourced.gemini.Hash (Hash.scala:86) - Extracting features
 WARN 19:13:42 tech.sourced.gemini.Hash (Hash.scala:103) - creating document frequencies
 WARN 19:13:44 tech.sourced.gemini.Hash (Hash.scala:122) - hashing features     
 WARN 19:13:44 tech.sourced.gemini.Gemini (Gemini.scala:45) - Saving hashes to DB
 WARN 19:13:44 tech.sourced.gemini.Hash (Hash.scala:148) - save meta to DB
 WARN 19:13:44 tech.sourced.gemini.Hash (Hash.scala:142) - save document frequencies to docf
req.json
 WARN 19:13:44 tech.sourced.gemini.Hash (Hash.scala:167) - save hashtables to DB
Done
```

Unfortunately when I run `report` I don't see any similar files.

```bash
$ docker-compose exec gemini ./report
Reporting all similar files
No duplicates found.
No similar files found.
```

It is possible there's none, or that there is an issue. And I can't really tell because I don't know which repositories *should* report similar files.

Could you check whether this is an actual issue or simply these organization do not contain any similar files (which seem improbable)?

Thanks

**Comment by bzz** [comment]
@campoy sure, will be happy to double-check this for you, if you provide either access to the dataset that you are running on (ssh to GCP), or a command that allows to get the data locally.

**Comment by bzz** [comment]
Quick glance shows that Google has 5008 siva files of 17Gb, Docker has 80 of 1.4G 

Numbers got from 
```
pga list -u github.com/google/ -f json |  jq -r '.sivaFilenames[]' | wc -l
pga list -u github.com/google/ -f json |  jq -r '.sivaFilenames[]' | pga get -j 90 -i -o googl
```

In my experience, this is way beyond the scenarios that `docker-compose` can be useful for and is not very realistic. To make that happen, it will require a lot of manual labor tuning every part of the process.

For the reference - you can see exact timings of every stage of Gemini that we [have collected](https://github.com/src-d/gemini/issues/42#issuecomment-396575463) in order to set performance expectations, as part of the initial release of Gemini by the end of quarter, using:
 - smaller dataset of just 10Gb
 - on Apache Spark cluster of 3 machines, 16Gb/32 cores for each worker 

TL;DR It is order of ~1h for the first stages that just use Engine to extract UASTs even on distributed configuration

**Comment by bzz** [comment]
@campoy  To answer original question on example of using Gemini on PGA, you would need to run it like:

```
docker-compose exec gemini ./hash ./repositories/siva/latest/
```

Reason is -  Engine only supports one level of nested dirs, so it does not find any .siva files but Gemini in this case provides confusing user output by listing .siva files recursively.

**Comment by bzz** [comment]
Closing this for now, as both, examples were provided and confusing CLI UX was fixed in #143 

But please, feel free to re-open in case something is not solved.

**Comment by campoy** [comment]
Why does the engine only support one level of nested directories?

**Comment by bzz** [comment]
> Why does the engine only support one level of nested directories?

IDK, but guess that may be a new issue in https://github.com/src-d/engine might be a better place to ask.

For your convenience, quick search over engine repo brings what looks like original issue that brought that feature https://github.com/src-d/engine/issues/176

**Comment by bzz** [comment]
Closing, if there is no further discussion
